MediaBrowser.Common/Providers/ProviderIdParsers.cs

```diff
@@ -0,0 +1,119 @@
+﻿#nullable enable
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace MediaBrowser.Common.Providers
+{
+    /// <summary>
+    /// Parsers for provider ids.
+    /// </summary>
+    public static class ProviderIdParsers
+    {
+        /// <summary>
+        /// Parses an IMDb id from a string.
+        /// </summary>
+        /// <param name="text">The text to parse.</param>
+        /// <param name="imdbId">The parsed IMDb id.</param>
+        /// <returns>True if parsing was successful, false otherwise.</returns>
+        public static bool TryParseImdbId(string text, [NotNullWhen(true)] out string? imdbId)
```

Comment:
I wrote this after the benchmark test. Dunno if it's more confusing, better or the same...

```C#
        public static string ExtractImdbId(string text)
        {
            var asSpan = text.AsSpan();
            var tt = "tt".AsSpan();

            while (asSpan.Length >= 9)
            {
                var ttPos = asSpan.IndexOf(tt);
                if (ttPos == -1)
                {
                    return null;
                }

                asSpan = asSpan.Slice(ttPos + tt.Length);
                var i = 0;
                for (; i < Math.Min(asSpan.Length, 9); i++)
                {
                    var c = asSpan[i];
                    if (!IsDigit(c))
                    {
                        break;
                    }
                }

                if (i <= 8 && i >= 7)
                {
                    return asSpan.Slice(0, i).ToString();
                }

                asSpan = asSpan.Slice(i);
            }
            
            return null;
        }

        private static bool IsDigit(char c)
        {
            return c >= '0' && c <= '9';
        }
```